iroh/src/commands/provide.rs

```diff
@@ -175,6 +175,7 @@ async fn get_keypair(key: Option<PathBuf>) -> Result<Keypair> {
     }
 }
 
+/// Makes a an RPC endpoint that uses a QUIC transport
 fn make_rpc_endpoint(
```

Comment:
went on a wild goose chase looking for a way to construct an RPC endpoint to give to the builder, and at one point wanted to move this, only to discover `node.controller()`. This comment is a momento of hour(s) lost.

Reply:
Sorry about this. So the idea is that you always want a cheap in memory way to control your node via RPC like calls.

Basically like an in memory message channel between you and the node, but nicer than the usual pattern of sending messages that contain a oneshot sender for the reply. This is what the controller is for.